### ISSUE TITLE
google_artifact_registry_repository_iam_member : location argument required for "apply" but not checked during "plan".

### ISSUE THREAD

**Issue by renaudguerin** [hashicorp/terraform-provider-google#13585]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v1.3.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.49.0
```

### Affected Resource(s)

* google_artifact_registry_repository_iam_member

### Terraform Configuration Files

```tf
resource "google_artifact_registry_repository_iam_member" "example_registry_access" {
  repository = google_artifact_registry_repository.repository.name
  # location   = google_artifact_registry_repository.repository.location
  role   = "roles/artifactregistry.writer"
  member = "serviceAccount:${google_service_account.someaccount.email}"
}
```

### Expected Behavior

The documentation [correctly states](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository_iam#argument-reference) that the `location` argument is required.
`terraform plan` should fail if it's missing.

### Actual Behavior

`terraform plan` succeeds even when the `location` argument is omitted in the resource block.
`terraform apply` then fails with : 
`Error: Error retrieving IAM policy for artifactregistry repository "projects/artifacts-xxxxxx/locations//repositories/yyyyyyy": googleapi: Error 400: Request contains an invalid argument.`

### Steps to Reproduce

1. `terraform apply` a successfully planned configuration that omits the `location` argument for `google_artifact_registry_repository_iam_member`

### References

The proper fix might actually be to determine `location` directly from the repository instead of making it a required argument ?
See #10436

b/272798254

**Comment by edwardmedia** [comment]
@renaudguerin does https://github.com/hashicorp/terraform-provider-google/issues/12989 help explain?

**Comment by renaudguerin** [comment]
> @renaudguerin does #12989 help explain?

Thanks for this. Not really, I still have the same question as the OP :

> As location is a required field, shouldn't it raise an error when the value is missing?

**Comment by renaudguerin** [comment]
Fixed in #14596
